core/src/main/java/org/apache/spark/ExecutorPluginContext.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark;
+
+import com.codahale.metrics.MetricRegistry;
+import org.apache.spark.annotation.DeveloperApi;
+
+/**
+ * This class encapsulates the input for the ExecutorPlugin interface. It provides a
+ * layer of indirection to ease future extensions of executor plugins, without breaking the API.
+ */
+@DeveloperApi
+public class ExecutorPluginContext {
+
+  public MetricRegistry executorPluginMetricRegistry;
```

Comment:
I have restructed ExecutorPluginContext, let me know if it has improved.

The idea of using the plugin name as source name is interesting and worth further discussing.  At present the code blocks the namespace=executorPlugin, which is more rigid, but easier to track. I see pros and cons to both approaches. I guess the preferred solution depends on how we expect users to make use of this functionality.

Reply:
I also feel like letting plugins choose their own namespace seems better, though I don't feel too strongly about this.

Reply:
OK. I can see multiple options, for example:

- fixed namespace/sourceName, user defines only metric name (current code)
- user/plugin allowed to specify sourceName too
- sourceName is set to plugin name
- sourceName is built by composing a fixed prefix + a user_specified part

Any preferences and/or other ideas?

Reply:
My preference would be either 3 or 4. 3 has the advantage that you can keep the current API and do that later without having to change the API (it would only change how the metrics are reported). 4 needs 3 (the prefix should be the plugin name) and would only be useful for plugins that want to register multiple sources, for whatever reason.

Reply:
OK for option 3 (sourceName is set to plugin name).